javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -11,39 +11,43 @@
 
 import java.io.Serializable;
 import java.util.*;
-import java.util.Set;
 import java.util.function.*;
 import java.util.stream.Collector;
 
+import static javaslang.collection.BitMappedTrie.lastDigit;
 import static javaslang.collection.Collections.*;
 
 /**
- * Vector is the default Seq implementation. It provides the best performance in between Array (with constant time element access)
- * and List (with constant time element addition).
+ * Vector is the default Seq implementation that provides effectively constant time access to any element.
+ * Many other operations (e.g. `tail`, `drop`, `slice`) are also effectively constant.
+ * <p>
+ * It's implemented using a `bit-mapped trie`, i.e. a tree where each node has `32` children.
+ * For integer indices this means that the tree's depth will be ≤ 6.
  *
  * @param <T> Component type of the Vector.
- * @author Ruslan Sennov
- * @since 2.0.0
+ * @author Ruslan Sennov, Pap Lőrinc
+ * @since 2.0.0, 2.1.0
```

Comment:
We do not protocal updates. Just leave only 2.0.0 please - the interface will remain the same. Changes to implementations are continuous.